wagtail/utils/deprecation.py

```diff
@@ -7,3 +10,50 @@ class RemovedInWagtail15Warning(DeprecationWarning):
 
 class RemovedInWagtail16Warning(PendingDeprecationWarning):
     pass
+
+
+class ThisShouldBeAList(list):
+    """
+    Some properties - such as Indexed.search_fields - used to be tuples. This
+    is incorrect, and they should have been lists.  Changing these to be a list
+    now would be backwards incompatible, as people do
+
+    .. code-block:: python
+
+        search_fields = Page.search_fields + (
+            SearchField('body')
+        )
+
+    Adding a tuple to the end of a list causes an error.
+
+    This class will allow tuples to be added to it, as in the above behaviour,
+    but will raise a deprecation warning if someone does this.
+    """
+    message = '{name} should be a list, not a {type}'
```

Comment:
The usual tone for deprecation warnings in Wagtail is usually something more on the lines of: "Setting {name} to a {type} is deprecated, please use a list instead."